src/target/adiv5_swd.c

```diff
@@ -344,8 +347,12 @@ uint32_t firmware_swdp_error(adiv5_debug_port_s *dp, const bool protocol_recover
 		 * into the expected state.
 		 */
 		swd_line_reset_sequence(true);
-		if (dp->version >= 2U)
-			firmware_dp_low_write(ADIV5_DP_TARGETSEL, dp->targetsel);
+		dp->fault = 0;
```

Comment:
This change is not required. This code already works and the fault state properly reset below. Please revert this change.

Reply:
As pointed out before, write to ADIV5_DP_TARGETSEL by design does not give an ack. On cmsis_dap < v1.2 by using a normal register access, we provoke an error that we than must recover from. Using a raw packet access without checking ack does not provoke the error.

Reply:
This code isn't even used on CMSIS-DAP - that's `dap_swd_clear_error()` that is. It's wrong to be modifying the firmware to fix that as this doesn't change the code path in question.